### PR TITLE
Add LockRoles extension

### PR DESCRIPTION
**Pull request by NickCuso**
# Description

An extension mixin to make it easier to interface with various roles from the lock.

# Issues

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contract-extensions/contracts/mixins/LockRoles.sol [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ) internal view
+    returns (bool)
+  {
+    address keyManager = _lock.keyManagerOf(_tokenId);
+    if(keyManager != address(0))
+    {
+      return keyManager == _user;
+    }
+    return _lock.isKeyOwner(_tokenId, _user);
+  }
```
Comment (nfurfaro) [inline smart-contract-extensions/contracts/mixins/LockRoles.sol:50]: Could we use the existing `_isKeyManager()` function from `MixinKeys.sol` here?

File: smart-contract-extensions/contracts/mixins/LockRoles.sol [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  /**
+   * @notice The key manager role is required to transfer or cancel a key.
```
Comment (nfurfaro) [inline smart-contract-extensions/contracts/mixins/LockRoles.sol:53]: An `approved` address should have this right also, correct?

File: smart-contract-extensions/contracts/mixins/LockRoles.sol (hunk 1 above)
Comment (NickCuso) [reply smart-contract-extensions/contracts/mixins/LockRoles.sol:50]: Yea, longer term I think we should change that to public.

File: smart-contract-extensions/contracts/mixins/LockRoles.sol (hunk 2 above)
Comment (NickCuso) [reply smart-contract-extensions/contracts/mixins/LockRoles.sol]: I removed this modifier. We don't have a use case for it yet and it was getting complicated.

**Pull request merged**